File: wrappers/java/src/test/java/org/hyperledger/indy/sdk/anoncreds/ProverGetCredentialTest.java
package org.hyperledger.indy.sdk.anoncreds;


import org.hyperledger.indy.sdk.wallet.WalletItemNotFoundException;
import org.json.JSONObject;
import org.junit.Test;

import java.util.concurrent.ExecutionException;

import static org.hamcrest.CoreMatchers.isA;
import static org.junit.Assert.assertTrue;

import org.hyperledger.indy.sdk.JsonObjectSimilar;

public class ProverGetCredentialTest extends AnoncredsIntegrationTest {

	@Test
	public void testProverGetCredentialWorks() throws Exception {
		String credentialJson = Anoncreds.proverGetCredential(wallet, credentialId1).get();

		JSONObject credential = new JSONObject(credentialJson);

		JSONObject expected = new JSONObject();
		expected.put("schema_id", gvtSchemaId);
		expected.put("cred_def_id", issuer1gvtCredDefId);
		expected.put("referent", credentialId1);
		expected.put("rev_reg_id", JSONObject.NULL);
		expected.put("cred_rev_id", JSONObject.NULL);
		JSONObject attrs = new JSONObject();
		attrs.put("sex", "male");
		attrs.put("name", "Alex");
		attrs.put("height", "175");
		attrs.put("age", "28");
		expected.put("attrs", attrs);

		assertTrue(JsonObjectSimilar.similar(expected, credential));
	}

	@Test
	public void testProverGetCredentialWorksForNotFound() throws Exception {
		thrown.expect(ExecutionException.class);
		thrown.expectCause(isA(WalletItemNotFoundException.class));

		Anoncreds.proverGetCredential(wallet, "other_cred_id").get();
	}

}

File: wrappers/dotnet/indy-sdk-dotnet-test/AnonCredsTests/ProverGetCredentialTest.cs
using Hyperledger.Indy.AnonCredsApi;
using Hyperledger.Indy.WalletApi;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using Newtonsoft.Json.Linq;
using System.Threading.Tasks;

namespace Hyperledger.Indy.Test.AnonCredsTests
{
    [TestClass]
    public class ProverGetCredentialTest : AnonCredsIntegrationTestBase
    {
        [TestMethod]
        public async Task TestProverGetCredentialWorks()
        {
            var credentialJson = await AnonCreds.ProverGetCredentialAsync(wallet, credentialId1);

            var credential = JObject.Parse(credentialJson);

            var expected = JObject.FromObject(new
            {
                schema_id = gvtSchemaId,
                cred_def_id = issuer1gvtCredDefId,
                referent = credentialId1,
                rev_reg_id = (object)null,
                cred_rev_id = (object)null,
                attrs = new
                {
                    sex = "male",
                    name = "Alex",
                    height = "175",
                    age = "28",
                }
            });

            Assert.IsTrue(JToken.DeepEquals(expected, credential));
        }

        [TestMethod]
        public async Task TestProverGetCredentialWorksForNotFound()
        {
            var ex = await Assert.ThrowsExceptionAsync<WalletItemNotFoundException>(() =>
                AnonCreds.ProverGetCredentialAsync(wallet, "other_cred_id")
            );
        }
    }
}
